packages/engine/Source/DataSources/FeatureLayer.js

```diff
@@ -0,0 +1,213 @@
+import Rectangle from "../Core/Rectangle.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import defined from "../Core/defined.js";
+import PolygonGeometry from "../Core/PolygonGeometry.js";
+import Feature from "./Feature.js";
+import AssociativeArray from "../Core/AssociativeArray.js";
+import CustomDataSource from "./CustomDataSource.js";
+import parseGeoJson from "./parseGeoJson.js";
+import Ellipsoid from "../Core/Ellipsoid.js";
+import Cartesian3 from "../Core/Cartesian3.js";
+
+/**
+ * @import FeatureProvider from "./FeatureProvider";
+ * @import FrameState from "../Scene/FrameState";
+ * @import OGCFeaturesProvider from "./OgcFeatureProvider.js";
+ */
+
+/**
+ * A class for things
+ * @constructor
+ * @param {FeatureProvider | OGCFeaturesProvider} provider
+ * @param {Object} options
+ */
+function FeatureLayer(provider, options) {
+  this.provider = provider;
+
+  this.id = options.id;
+  // this.name = options.name;
+  this._show = options.show ?? true;
+
```

Comment:
Since this defines a `Rectangle`, we should likely be specifying an ellipsoid at this granularity as well so it's meaningful.